src/main/java/org/prebid/server/bidder/ix/IxBidder.java

```diff
@@ -194,21 +201,48 @@ private static Publisher modifyPublisher(Publisher publisher, String siteId) {
         }
     }
 
-    private static List<BidderBid> extractBids(BidResponse bidResponse, BidRequest bidRequest) {
+    private List<BidderBid> extractBids(BidResponse bidResponse, BidRequest bidRequest) {
         return bidResponse == null || CollectionUtils.isEmpty(bidResponse.getSeatbid())
                 ? Collections.emptyList()
                 : bidsFromResponse(bidResponse, bidRequest);
     }
 
-    private static List<BidderBid> bidsFromResponse(BidResponse bidResponse, BidRequest bidRequest) {
+    private List<BidderBid> bidsFromResponse(BidResponse bidResponse, BidRequest bidRequest) {
         return bidResponse.getSeatbid().stream()
                 .map(SeatBid::getBid)
                 .flatMap(Collection::stream)
-                .map(bid -> prepareBid(bid, bidRequest))
-                .map(bid -> BidderBid.of(bid, getBidType(bid.getImpid(), bidRequest.getImp()), bidResponse.getCur()))
+                .map(bid -> toBidderBid(bid, bidRequest, bidResponse))
                 .collect(Collectors.toList());
     }
 
+    private BidderBid toBidderBid(Bid bid, BidRequest bidRequest, BidResponse bidResponse) {
+        final BidType bidType = getBidType(bid.getImpid(), bidRequest.getImp());
+
+        final boolean bidHasNoSizes = bid.getH() == null || bid.getW() == null;
+        final Banner banner = bidRequest.getImp().get(0).getBanner();
+        if (bidHasNoSizes && banner != null) {
+            bid = bid.toBuilder()
+                    .w(banner.getW())
+                    .h(banner.getH())
+                    .build();
+        }
+
+        final ExtBidPrebid bidExt = parseBidExt(bid.getExt());
+        if (bidExt != null && bidExt.getVideo() != null) {
+            final ExtBidPrebidVideo video = bidExt.getVideo();
+            bid = bid.toBuilder()
+                    .ext(toBidderBidExt(video))
+                    .build();
+            if (bid.getCat().size() == 0) {
```

Comment:
CollectionUtils.isEmpty(bid.getCat())

Reply:
Why not to use CollectionUtils.isEmpty(bid.getCat())?

Reply:
Done